providers/nutshell/connector.go

```diff
@@ -26,5 +33,34 @@ func constructor(base *components.Connector) (*Connector, error) {
 
 	connector.SchemaProvider = schema.NewOpenAPISchemaProvider(connector.ProviderContext.Module(), metadata.Schemas)
 
+	errorHandler := interpreter.ErrorHandler{
+		JSON: interpreter.NewFaultyResponder(errorFormats, statusCodeMapping),
+		HTML: &interpreter.DirectFaultyResponder{Callback: connector.interpretHTMLError},
+		Custom: map[interpreter.Mime]interpreter.FaultyResponseHandler{
+			"text":       textErrorStatusCodeHandler,
+			"text/plain": textErrorStatusCodeHandler,
+		},
+	}.Handle
+
+	connector.Reader = reader.NewHTTPReader(
+		connector.HTTPClient().Client,
+		components.NewEmptyEndpointRegistry(),
+		connector.ProviderContext.Module(),
+		operations.ReadHandlers{
+			BuildRequest:  connector.buildReadRequest,
+			ParseResponse: connector.parseReadResponse,
+			ErrorHandler:  errorHandler,
+		},
+	)
+
 	return connector, nil
 }
+
+func (c *Connector) getReadURL(objectName string) (*urlbuilder.URL, error) {
+	objectPath, err := metadata.Schemas.FindURLPath(common.ModuleRoot, objectName)
+	if err != nil {
+		return nil, err
+	}
+
+	return urlbuilder.New(c.ModuleInfo().BaseURL, apiURIPart, objectPath)
```

Comment:
since there are no modules, it would read easier if this was c.ProviderInfo().BaseURL, but feel free to ignore